infrastructure/tooling/src/smoketest/checks/role.js

```diff
@@ -27,6 +27,16 @@ exports.tasks.push({
         'project:taskcluster:smoketest:abc/*' ],
     };
     assert.deepEqual(expandedRole.scopes, expectedScopes.scopes);
-    await auth.deleteRole(roleId);
+    const query = {limit: 1};
```

Comment:
While using `limit: 1` is good for testing the continuation support, it's not very efficient, so now that you're sure the continuation support works, remove the limit.  That will also require looping over the roles returned from `listRoles2`.

Reply:
I am a little confused regarding this part. Can you please check my latest pushed code and tell me the changes that I can make to get the clean up to work correctly?